src/main/java/frc/robot/utils/InterpolateDouble.java

```diff
@@ -0,0 +1,39 @@
+package frc.robot.utils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+
+public class InterpolateDouble {
+    private HashMap<Double, Double> map;
+
+    public InterpolateDouble(HashMap<Double, Double> map) {
+        this.map = map;
+    }
+
+    public double getValue(double key) {
+        if (map.containsKey(key)) {
+            return map.get(key);
+        }
+
+        ArrayList<Double> sortedKeys = new ArrayList<Double>(map.keySet());
+        Collections.sort(sortedKeys);
```

Comment:
It could be good to do this in the constructor to avoid calling `sort` every time we want to get a value